native/index.ts

```diff
@@ -230,27 +242,60 @@ app.on("will-quit", () => {
   proxyProcessManager.kill();
 });
 
-// This method will be called when Electron has finished
-// initialization and is ready to create browser windows.
-// Some APIs can only be used after this event occurs.
-app.on("ready", () => {
-  proxyProcessManager.run().then(({ status, signal, output }) => {
-    windowManager.sendMessage({
-      kind: MainMessageKind.PROXY_ERROR,
-      data: {
-        status,
-        signal,
-        output,
-      },
-    });
+// Handle custom protocol on macOS
+app.on("open-url", (event, url) => {
+  event.preventDefault();
+
+  windowManager.sendMessage({
+    kind: MainMessageKind.CUSTOM_PROTOCOL_INVOCATION,
+    data: { url },
+  });
+});
+
+if (app.requestSingleInstanceLock()) {
+  // Handle custom protocol on Linux when Upstream is already running
+  app.on("second-instance", (_event, commandLine, _workingDirectory) => {
+    windowManager.focus();
+    if (commandLine[1] && commandLine[1].toLowerCase().match(/^radicle:\/\//)) {
+      windowManager.sendMessage({
+        kind: MainMessageKind.CUSTOM_PROTOCOL_INVOCATION,
+        data: { url: commandLine[1] },
+      });
+    }
```

Comment:
This block is duplicated

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1652/commits/fcc8543f16bc5a902b0b82ddc05824d4b98d2c9d